### DISCUSSION TITLE
Setting files in FileUpload

### DISCUSSION THREAD

**Discussion by XOP** [chakra-ui/ark#2701] [Q&A]
Hello team!
Thanks for all the great work on Ark UI, really enjoy it!

I've a question about FileUpload - in version 1.3 it was possible to set files via `files` prop,   
however it's no longer possible in version 3.x (didn't check 2.x)

What is the recommended solution for this case? Couldn't find respective documentation.   
Thank you!

**Comment by cschroeter** [comment]
`acceptedFiles` is a read-only field. A way to do this is to merge the accepted files with outsides files

```jsx
<FileUpload.ItemGroup>
  <FileUpload.Context>
    {({ acceptedFiles }) =>
      [...acceptedFiles, ..otherFiles].map((file) => (
        <FileUpload.Item key={file.name} file={file} />
      ))
    }
  </FileUpload.Context>
</FileUpload.ItemGroup>
```

**Reply by dhay** [reply]
I'm not sure I understand how this can be used to control the set of uploaded files?  After submitting my form, I want to clear the set of uploaded files, but it's non-obvious to me how to do that based on the documentation [here](https://www.chakra-ui.com/docs/components/file-upload) and attempting to read through the code.
